internal/core/processor/evolution/client.go

```diff
@@ -0,0 +1,141 @@
+package evolution
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"net/http"
+)
+
+const (
+	jsonContentType = "application/json"
+	postMethod      = "POST"
+)
+
+// Define an interface for HTTP client operations
+type LaosRPCRequests interface {
+	LatestFinalizedBlockHash() (string, error)
+	BlockNumber(blockHash string) (*big.Int, error)
+}
+
+// Define an interface for HTTP client operations
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type LaosHTTP struct {
+	Client HTTPClient
+	Url    string
+}
+
+func NewLaosHTTP(url string) LaosRPCRequests {
```

Comment:
you should consider dependency injection here for HTTPClient (we use functional option pattern for this)